geonode/api/authorization.py

```diff
@@ -30,11 +29,17 @@ class GeoNodeAuthorization(DjangoAuthorization):
     permission system"""
 
     def read_list(self, object_list, bundle):
-        permitted_ids = get_objects_for_user(
+        permitted_view_ids = get_objects_for_user(
             bundle.request.user,
-            'base.view_resourcebase').values('id')
+            'base.view_resourcebase').values_list(
+            'id',
+            flat=True)
+        return object_list.filter()
```

Comment:
We intentionally deleted the filter here, in order to show the whole catalogue for every user.